x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/AbstractFunctionTestCase.java

```diff
@@ -642,6 +642,46 @@ public interface ExpectedEvaluatorToString {
         Matcher<String> evaluatorToString(int nullPosition, TestCaseSupplier.TypedData nullData, Matcher<String> original);
     }
 
+    /**
+     * Modifies suppliers to generate BytesRefs with random offsets.
+     */
+    protected static List<TestCaseSupplier> randomizeBytesRefsOffset(List<TestCaseSupplier> testCaseSuppliers) {
```

Comment:
Wasn't sure about pushing this in this PR, but as it discovered another bug, well...

My idea would be to propagate it to other tests (Or at least to new functions)

Reply:
:+1: on separating it it for now.

I wonder if we could merge this into `TypedData`'s constructor. It already has a funny hack for `UNSIGNED_LONG`. I wonder if we can do a hack for `BytesRef` and randomize the offset there.

Reply:
Ok! Extracting to another branch, and merging

Reply:
About the TypedData constructor, I would avoid it, as I could imagine some test manually adding offsets themselves for some usecase (?)
Not sure, but I feel like there's more control this way